node_modules/oae-principals/lib/internal/dao.js

```diff
@@ -402,6 +361,87 @@ var isUser = module.exports.isUser = function(userId) {
 };
 
 /**
+ * Perform a raw update with the given profile fields without any checking if restricted fields are
+ * being set, and ensuring the cache remains in a valid state
+ *
+ * @param  {String}     principalId     The id of the principal to update
+ * @param  {Object}     profileFields   An object containing the profile field updates to apply
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ * @api private
+ */
+var _updatePrincipal = function(principalId, profileFields, callback) {
+    // Take out the `principalId` to avoid setting a partial cache entry. Also ensure we have a
+    // `lastModified` set for the update
+    profileFields = _.chain({})
+        .defaults(profileFields, {'lastModified': Date.now().toString()})
+        .omit('principalId')
+        .value();
+
+    // Ensure we aren't updating a non-principal to avoid upserting invalid rows
+    var validator = new Validator();
+    validator.check(principalId, {'code': 400, 'msg': 'Attempted to update a principal with a non-principal id'}).isPrincipalId();
+    if (validator.hasErrors()) {
+        return callback(validator.getFirstError());
+    }
+
+    // Create the cassandra query from object fields
+    var q = Cassandra.constructUpsertCQL('Principals', 'principalId', principalId, profileFields);
+    if (!q) {
+        return callback({'code': 500, 'msg': 'Unable to update profile fields'});
+    }
+
+    // Perform the update
+    Cassandra.runQuery(q.query, q.parameters, function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        // Update the cache, if necessary
+        return OaeUtil.invokeIfNecessary(isUser(principalId), _updateCachedUser, principalId, profileFields, callback);
+    });
+};
+
+/**
+ * Remove the specified fields from the principal
+ *
+ * @param  {String}     principalId     The id of the principal whose fields to delete
+ * @param  {String[]}   profileFields   The names of the fields to delete
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var _deletePrincipalFields = function(principalId, profileFields, callback) {
+    // Remove the specified fields
+    var query = util.format('DELETE "%s" FROM "Principals" where "principalId" = ?', profileFields.join('", "'));
+    Cassandra.runQuery(query, [principalId], function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        // If the principal is a user, invalidate their cache entry
+        OaeUtil.invokeIfNecessary(isUser(principalId), invalidateCachedUsers, [principalId], function(err) {
```

Comment:
At first I thought this line could be removed because we're calling `_updatePrincipal` which does some cache handling.
However, that isn't true as we're removing principal fields here and `_updatePrincipal` will only update specific profile fields, not refresh the entire object.

So, could you add a bit of documentation why `invalidateCachedUsers` is required here?